amplified_todo/lib/pages/TodosPage.dart

```diff
@@ -55,22 +55,19 @@ class _TodosPageState extends State<TodosPage> {
     // configure Amplify
     await _configureAmplify();
 
-    // listen for updates to Todo entries by passing the Todo classType to
-    // Amplify.DataStore.observe() and when an update event occurs, fetch the
-    // todo list
+    // Query and Observe updates to Todo models. DataStore.observeQuery() will
+    // emit an initial QuerySnapshot with a list of Todo models in the local store,
+    // and will emit subsequent snapshots as updates are made
     //
-    // note this strategy may not scale well with larger number of entries
-    _subscription = Amplify.DataStore.observe(Todo.classType).listen((event) {
-      _fetchTodos();
-    });
-
-    // fetch Todo entries from DataStore
-    await _fetchTodos();
-
-    // after both configuring Amplify and fetching Todo entries, update loading
-    // ui state to loaded state
-    setState(() {
-      _isLoading = false;
+    // each time a snapshot is received, the following will happen:
+    // isLoading is set to false if it is not already false
+    // todos is set to the value in the latest snapshot
+    _subscription = Amplify.DataStore.observeQuery(Todo.classType)
+        .listen((QuerySnapshot<Todo> snapshot) {
+      setState(() {
+        if (_isLoading) _isLoading = false;
```

Comment:
Nit: Is the check technically even necessary?

Reply:
Without it there will be a period of time after the app starts where there will be an empty list of todos. Then once the first snapshot emits, the list will be updated.